src/dlangui/graphics/glsupport.d

```diff
@@ -41,37 +41,43 @@ version (Android) {
 
     static if (SUPPORT_LEGACY_OPENGL) {
         public import GLES.gl : glEnableClientState, glLightfv, glColor4f, GL_ALPHA_TEST, GL_VERTEX_ARRAY,
-		    GL_COLOR_ARRAY, glVertexPointer, glColorPointer, glDisableClientState,
-		    GL_TEXTURE_COORD_ARRAY, glTexCoordPointer, glColorPointer, glMatrixMode,
-		    glLoadMatrixf, glLoadIdentity, GL_PROJECTION, GL_MODELVIEW;
-	}
+            GL_COLOR_ARRAY, glVertexPointer, glColorPointer, glDisableClientState,
+            GL_TEXTURE_COORD_ARRAY, glTexCoordPointer, glColorPointer, glMatrixMode,
+            glLoadMatrixf, glLoadIdentity, GL_PROJECTION, GL_MODELVIEW;
+    }
 
 } else {
     enum SUPPORT_LEGACY_OPENGL = false; //true;
-    public import derelict.opengl;
-    //public import derelict.opengl.types;
-    //public import derelict.opengl.versions.base;
-    //public import derelict.opengl.versions.gl3x;
-    //public import derelict.opengl.gl;
+    public import bindbc.opengl;
+    public import bindbc.opengl.bind.arb.core_30;
+    import loader = bindbc.loader.sharedlib;
+
 
-derelict.util.exception.ShouldThrow gl3MissingSymFunc( string symName ) {
+private void gl3CheckMissingSymFunc(const(loader.ErrorInfo)[] errors)
+{
     import std.algorithm : equal;
-    static import derelict.util.exception;
-    foreach(s; ["glGetError", "glShaderSource", "glCompileShader",
+    immutable names = ["glGetError", "glShaderSource", "glCompileShader",
             "glGetShaderiv", "glGetShaderInfoLog", "glGetString",
             "glCreateProgram", "glUseProgram", "glDeleteProgram",
             "glDeleteShader", "glEnable", "glDisable", "glBlendFunc",
             "glUniformMatrix4fv", "glGetAttribLocation", "glGetUniformLocation",
             "glGenVertexArrays", "glBindVertexArray", "glBufferData",
-            "glBindBuffer", "glBufferSubData"]) {
-        if (symName.equal(s)) // Symbol is used
-            return derelict.util.exception.ShouldThrow.Yes;
+            "glBindBuffer", "glBufferSubData"];
+    foreach(info; errors)
+    {
+        import std.array;
+        import std.algorithm;
+        import std.exception;
+        import core.stdc.string;
+        // NOTE: this has crappy complexity as it was just updated as is
+        //     it also does not checks if the symbol was actually loaded
+        auto errMsg = cast(string) info.message[0 .. info.message.strlen];
+        bool found = names
```

Comment:
I guess this code is wrong because it finds `errMsg` in `names` instead of `names` in `errMsg`. Also using `array` to count items is a bad practice.
```suggestion
    bool found = any!(s=>errMsg1.canFind(s))(names);
```